### PR TITLE
Zombies can now succumb

### LINKED ISSUE #70877: Halloween high functioning zombies can't succumb emprisioning them in a eternal crit state unless ghosted
## Round ID:
[193309](https://scrubby.melonmesa.com/round/193309)

## Reproduction:
Halloween high functioning zombies can't succumb emprisioning them in a eternal crit state unless ghosted

### PR DESCRIPTION
**Pull request by Time-Green**
fixes #70877

Mobs with NODEATH can now succumb if they choose. The only mobs with NODEATH are all zombies, memento mori and penthrite, of which zombies are the only one that can ever enter hardcrit. I think it's fair to let them die if they want.

High-functioning zombies can't succumb by whispering, but pressing the succumb UI without whispering will succumb you

:cl:
tweak: Mobs with NODEATH can now succumb (this is literally only zombies)
/:cl:

### CODE REVIEW

**Comment by LarissaMayrink** [comment]
thank god time, you my favorite coder

**Comment by Livrah** [comment]
Balance changes for 3 days event, nerf keeps entire year.

**Comment by LarissaMayrink** [comment]
> Balance changes for 3 days event, nerf keeps entire year.

What nerf? it only affects high functioning zombies, not romerol ones, and its for them to die instead of being stuck on permanent crit even on lava for half a hour

**Comment by Time-Green** [comment]
> Balance changes for 3 days event, nerf keeps entire year.

Maintainer called it a high-priority bug. High-functioning zombies are also halloween only (and an unused away-mission it looks like)

**Comment by Simplehorror** [comment]
swap it for no soft crit or something

**Comment by LarissaMayrink** [comment]
> swap it for no soft crit or something

why make them immune to soft crit? they are already strong enough given spaceproofing and no breath without any damage taken changes. All i saw in both testmerged zombie rounds on manuel were zombie tators raising terror. Making them stay up for longer is bad

**Comment by LarissaMayrink** [comment]
also there's only three more days, we gotta get this in

**Comment by Livrah** [comment]
> > Balance changes for 3 days event, nerf keeps entire year.
> 
> Maintainer called it a high-priority bug. High-functioning zombies are also halloween only (and an unused away-mission it looks like)

Pride Mirror and Abductor Surgery.

**Comment by the-orange-cow** [comment]
> > > Balance changes for 3 days event, nerf keeps entire year.
> > 
> > 
> > Maintainer called it a high-priority bug. High-functioning zombies are also halloween only (and an unused away-mission it looks like)
> 
> Pride Mirror and Abductor Surgery.

I think abductor surgery is romerol, but botanists can make zombie mut toxin with strange seeds and high instability mutation(rare).

**Comment by Time-Green** [comment]
> I think abductor surgery is romerol, but botanists can make zombie mut toxin with strange seeds and high instability mutation(rare).

Abductor surgery is romerol yeah. Zombie mut toxin is also admin restricted and can't be found in sources like mut toxin

**Comment by the-orange-cow** [comment]
> > I think abductor surgery is romerol, but botanists can make zombie mut toxin with strange seeds and high instability mutation(rare).
> 
> Abductor surgery is romerol yeah. Zombie mut toxin is also admin restricted and can't be found in sources like mut toxin

Within the past several months I've got it for high instability mutations.

**Comment by Time-Green** [comment]
> 

Apparently the list of blacklisted reagents are not actually blacklisted? Someone should unfuck that
https://github.com/tgstation/tgstation/blob/7fa3d3756b1272af23220e3eed7fc3f3717b2405/code/modules/reagents/chemistry/reagents/other_reagents.dm#L686-L709

Anyway I've misunderstood ShizCalev, I've changed the PR to just let zombies succumb if they want

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Comment by the-orange-cow** [comment]
> Apparently the list of blacklisted reagents are not actually blacklisted? Someone should unfuck that

Ummmmm... I take it back, I must have been hallucinating.  Plz no take funny toxins(
Its not like they hurt anyone).

File: code/__DEFINES/mobs.dm [hunk 1]
```diff
@@ -536,7 +536,7 @@
 #define SILENCE_RANGED_MESSAGE (1<<0)
 
 /// Returns whether or not the given mob can succumb
-#define CAN_SUCCUMB(target) (HAS_TRAIT(target, TRAIT_CRITICAL_CONDITION) && !HAS_TRAIT(target, TRAIT_NODEATH))
+#define CAN_SUCCUMB(target) (HAS_TRAIT(target, TRAIT_CRITICAL_CONDITION))
```
[the remark this answers is not in the document]
Comment (Time-Green) [reply code/__DEFINES/mobs.dm:539]: Oh you gotta be fucking kidding me. I’ll just seperate the fucker, cause UI succumb is intentional and whisper succumb isn’t

**Comment by LarissaMayrink** [comment]
revhead high functioning zombies also can never die thus never ending the revolution

**Comment by ATH1909** [comment]
> revhead high functioning zombies also can never die thus never ending the revolution

just behead or cremate them??

**Comment by ATH1909** [comment]
idk, I kind of like how memento mori and such can trap you in a fate worse than death. trait_nodeath means no death, even if it would be a sweet release from the hell you're going through.

hubris
hubris2
image

**Comment by ATH1909** [comment]
hubrischatlog

**Comment by Serxule** [comment]
> idk, I kind of like how memento mori and such can trap you in a fate worse than death. trait_nodeath means no death, even if it would be a sweet release from the hell you're going through.
> 

ashdrake (and all megafauna) are supposed to be able to gib you even with memento, thats a bug

**Comment by LarissaMayrink** [comment]
> idk, I kind of like how memento mori and such can trap you in a fate worse than death. trait_nodeath means no death, even if it would be a sweet release from the hell you're going through.

this doens't change anything about memento, only high funcitoning zombies tho

File: code/__DEFINES/mobs.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (Time-Green) [reply code/__DEFINES/mobs.dm:539]: Alright fixed. New trait: TRAIT_BLOCKED_WHISPER_SUCCUMB. High-functioning zombies can no longer succumb by whispering and must use the UI without whispering

**Comment by XDTM** [comment]
> > Apparently the list of blacklisted reagents are not actually blacklisted? Someone should unfuck that
> 
> Ummmmm... I take it back, I must have been hallucinating. Plz no take funny toxins( Its not like they hurt anyone).
> 
> Edit: Upon closer inspection, it seems as if the races are meant to be blacklisted(as in via roundstart from character customization), but not necessarily the toxins themselves(i.e. the potential to racechange under rare circumstances during the round). The races listed aren't particularly game breaking, so it makes sense that the toxins were never meant to be blacklisted.

In case you're wondering, they were blacklisted from being available from xenobio's mutation toxins, back when they used to change you to any existing species instead of the slimeperson subtypes.

File: code/__DEFINES/mobs.dm [hunk 2]
```diff
@@ -536,7 +536,8 @@
...
+#define CAN_WHISPER_SUCCUMB(target) (CAN_SUCCUMB(target) && !HAS_TRAIT(target, TRAIT_BLOCK_WHISPER_SUCCUMB))
```
Comment (Mothblocks) [inline code/__DEFINES/mobs.dm:540]: Document

File: code/modules/mob/living/living.dm [hunk 3]
```diff
@@ -432,9 +432,10 @@
 
 /mob/living/verb/succumb(whispered as null)
 	set hidden = TRUE
-	if (!CAN_SUCCUMB(src))
-		to_chat(src, text="You are unable to succumb to death! This life continues.", type=MESSAGE_TYPE_INFO)
+	if (whispered && !CAN_WHISPER_SUCCUMB(src))
+		to_chat(src, text="You cannot succumb while whispering!", type=MESSAGE_TYPE_INFO)
```
Comment (Mothblocks) [inline code/modules/mob/living/living.dm:436]: This message doesn't seem clear to me. I think you should keep the old one but display a special message if they have the rarer trait.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by Time-Green** [comment]
@Mothblocks cleaned shit up a bit. Also doesn't seem to break anything in normal succumb, zombie succumb and romerol zombie succumb, so thats a plus

Also I really, really hate to be this kind of person but I am but a poor feature coder
image

**Comment by Mothblocks** [comment]
ok but only because you asked nicely

**Pull request merged**